### PR TITLE
fix: debug store

### PR DESCRIPTION
**Pull request by dai-shi**
- [x] fix basic test
- fix onmount test -> follow-up pr
- fix quey/atomWithQuery test -> follow-up pr

### CODE REVIEW

File: src/core/Provider.ts [hunk 1]
```diff
@@ -60,9 +59,14 @@ const stateToPrintable = ([state, atoms]: [State, AnyAtom[]]) =>
 const isDevStore = (store: Store): store is StoreForDevelopment => {
   return store.length > 2
 }
-export const getDevState = (state: State) => ({ ...state }) // shallow copy XXX might be better ways
-export const getDevAtoms = ({ atoms }: { atoms: AnyAtom[] }) => atoms
-export const subscribeDevAtoms = (
+export const getDebugStateAndAtoms = ({
```
Comment (Thisen) [inline src/core/Provider.ts:62]: Instead of `debug*`, maybe call it `snapshot*`, because the atoms and their state is a snapshot?

**Review by Thisen — COMMENTED** [review]
LGTM.

File: src/core/Provider.ts (hunk 1 above)
Comment (dai-shi) [reply src/core/Provider.ts:62]: To me, snapshots are immutable. `atoms` are immutable, but `state` is not.

**Comment by dai-shi** [comment]
@Thisen
It looks like it would require more time on the remaining tasks.
As what we have now is probably better than `master`, do you think it's ok to merge this and tackle remaining issues separately?

**Review by Thisen — APPROVED** [review]

**Pull request merged**